recipe/meta.yaml

```diff
@@ -43,7 +43,7 @@ test:
     - datasets  # [not s390x]
     - expecttest
     - fsspec
-    - numpy
+    - {{ pin_compatible('numpy') }}
```

Comment:
isn't that pin_compatible has to be used with corresponding package from build section ?, from https://github.com/pytorch/data/blob/901b48312179865657b90792aa03f0775ec50971/test/requirements.txt#L6 seems could be any, but better to take first stable major release just in case so numpy => 1.0 ?

Reply:
I think `pin_compatible` doesn't hurt, however normal `numpy` should also work.

Reply:
I couldn't figure out why, but simply using `numpy` was giving me solver conflicts for Python 3.9, even though I could create an environment with all of the packages in test/run/host without issue. `pin_compatible` solved it for whatever reason.

Reply:
If you check the `host` environments for **py38**-**py311** you will find that `numpy` has versions:
- py38 - 1.24.3
- py39 - 1.25.0
- py310 - 1.25.0
- py311 - 1.25.0
The `test` envs always have the same `numpy` versions.
I'm not sure why a plain `numpy` gives conflicts